### DISCUSSION TITLE
Zwift not moving motor enough

### DISCUSSION THREAD

**Discussion by d-nx** [doudar/SmartSpin2k#223] [Q&A]
I just finished my build the other day and for the most part it's been working well.

*SS2K connects to my Echelon Sport and Zwift pairs to SS2k and I get good data from bike to ZWIFT (Watts / Cadence)
*Shifters work and move my motor correctly (UP=more resistance, DOWN=Less resistance)

My only problem is, when riding in Zwift, I'm not getting accurate info. My motor moves very little, and in what feels like very random intervals. Sometimes a little, sometimes a lot. Other times it starts moving even when my avatar is parked and Im not pedaling.

I'm hoping someone can point me in the right direction as to how to solve this.

This is the motor I got:
[Usongshine Nema 17 Stepper Motor Bipolar 38mm Body 42N.cm (60oz.in) 1.5A 4-Lead with 1m Cable and Connector for 3D Printer/Extruder/DIY CNC(17HS4401S)](https://www.amazon.com/gp/product/B07TZ4DPK1/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1)

I've also made sure I've adjusted the knob on my TMC2225 to the max position

Any help is greatly appreciated!

PS
I've got some cool octolapses of the prints if you want to use.

**Comment by doudar** [comment]
If the shifters are working correctly, all the hardware and wiring should be correct. What fork or binary, of the software are you using?

Increasing your incline multiplier may help.

**Reply by d-nx** [reply]
Thanks for the reply.

I'm not sure which version I'm running, but I flashed from the link that's on the root level instructions. My bike is a Echelon Sport and I'm trying to make it work with zwift running on my iMac.

I've maxed the incline sensitivity coming from zwift, and also the incline multiplier & steps on ss2k but still seeing that behaviour.

I'm happy to reflash with a different version if you think that would help.

Thanks again!

**Reply by doudar** [reply]
possibly your stepper driver enable pin isn't hooked up correctly?

I'm not certain what else could be causing this, but a brand new version 1.6.19 is hot off the presses.

It should auto update if you're connected to the internet, otherwise, you can find it (firmware.bin) here:
https://github.com/doudar/OTAUpdates/blob/main/firmware.bin

**Reply by d-nx** [reply]
Thanks. I'll give it a try Tomorrow and let you know if it's any better. After some more testing and monitoring the debugging console, I noticed that Zwift was taking a while in sending the incline commands. Not sure what would cause this, but I'll test with the new build and let you know!

**Reply by doudar** [reply]
Yah, normally Zwift should be sending incline commands every second. 

The only similar thing I've seen to this is Zwift has a (common but not always) bug where if you pause a workout in ERG mode, the ERG commands will get out of sync with the workout, delayed by however long the workout was paused.

**Reply by doudar** [reply]
The develop branch has a brand new way of generating motor steps using the dedicated MCPWM hardware on the ESP32. This should produce much smoother steps to your driver as well as reducing a lot of CPU time. Give it a try and let me know if it helps your issue. I should have a new release with it included in the upcoming days once I hear back from the testers. In the meantime, here's a binary with the current info. You may manually have to upload the compatible .html files from https://github.com/doudar/SmartSpin2k/tree/develop/data

[firmware.zip](https://github.com/doudar/SmartSpin2k/files/6727303/firmware.zip)

More info on MCPWM:
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/mcpwm.html

**Reply by d-nx** [reply]
Thanks for following up!

I've actually installed "1.3.21-develop-245-ge55a15e" because of the integration with "QZ". marklogan17 sent it to me and it's working well for my application.

I'm now following this post for any new info: https://github.com/cagnulein/qdomyos-zwift/discussions/377?fbclid=IwAR276lNHF0O_OEDBD3Vw1VbPHH3TfLn2vNxwksFWAXHUYmNqd2lHnn0DE44

Thanks again!
-D

**Reply by doudar** [reply]
Awesome! Great to know it's working for you!

**Comment by doudar** [comment]
Hey! I just found your issue. I think it's related to windows ble adaptors and Zwift. Not seeing it on any of the other apps. A workaround is to use companion app for connections. 

You'll probably have to quit Zwift or disable your windows BLE adapter for a second if you try to hot swap to companion app because Zwift doesn't close the connection on "unpair" 🤦 

If you don't have companion app, Unpairing, then disabling my windows BLE adaptor, then enabling it and re-pairing also worked.
 
Wahoo trainers (and probably others) seem to have the same issue with Zwift currently.

**Reply by d-nx** [reply]
Hmmm I wonder if it's the same on OSX since I'm running Zwift on an iMac

**Reply by doudar** [reply]
I believe it is: https://forums.zwift.com/t/wahoo-kickr-core-1-1-1-firmware-issues/551758/8

**Reply by doudar** [reply]
I'll mess around with specifying a different simulation grade range and see if it fixes it but it's not easily replicated on my machine.

**Reply by doudar** [reply]
@d-nx , give this firmware a try (current build of develop). We weren't broadcasting the inclination range characteristic before and the newest develop does. This has a chance of fixing this issue. In my testing I haven't seen it again. Again remember to unzip it and upload it as firmware.bin or use the programmer. 

[firmware.zip](https://github.com/doudar/SmartSpin2k/files/6817964/firmware.zip)
